### PR TITLE
Cleanup util

### PR DESCRIPTION
**Pull request by robkinyon**
This contains the first pass of code cleanup.

Please make sure to squash-merge this (and all future) PRs.

### CODE REVIEW

**Comment by singingfish** [comment]
Error:  Global symbol "$attrs" requires explicit package name at /home/travis/build/singingfish/braintree_perl/.build/TqcTpj9IbG/blib/lib/WebService/Braintree/CreditCard.pm line 106. :(

**Comment by singingfish** [comment]
I ran the sandbox tests, and became concerned about the number of failures.  Without going into where these regressed (previously I had assumed changes on the braintree end), I decided to have a look.  There's a bit of a rabbit hole here:

for `xt/sandbox/credit_card_verification_search.t` I fixed some stuff with this patch

    diff --git a/lib/WebService/Braintree/ResourceCollection.pm b/lib/WebService/Braintree/ResourceCollection.pm
    index 26fca0a..b20979f 100644
    --- a/lib/WebService/Braintree/ResourceCollection.pm
    +++ b/lib/WebService/Braintree/ResourceCollection.pm
    @@ -26,7 +26,7 @@ sub is_success {
     sub first {
         my $self = shift;
         return undef if $self->is_empty;
    -    my $first_id = $self->ids->[0];
    +    my $first_id = ref $self->ids eq 'ARRAY' ? $self->ids->[0] : $self->ids;
         return $self->callback->([$first_id])->[0];
     }
 
    @@ -37,7 +37,8 @@ sub is_empty {
     
     sub maximum_size {
         my $self = shift;
    -    return (scalar @{$self->ids});
    +    my @ids = ref($self->ids) eq 'ARRAY' ? @{$self->ids} : $self->ids;
    +    return (scalar @ids);
     }
 
But then failures cascade to:

    Single parameters to new() must be a HASH ref at constructor WebService::Braintree::CreditCardVerification::new (defined at lib/WebService/Braintree/CreditCardVerification.pm line 84) line 11

File: lib/WebService/Braintree/Result.pm [hunk 1]
```diff
@@ -2,7 +2,7 @@ package WebService::Braintree::Result;
 
 use Moose;
 use Hash::Inflator;
-use WebService::Braintree::Util;
+#use WebService::Braintree::Util;
```
Comment (singingfish) [inline lib/WebService/Braintree/Result.pm]: this line should be deleted

File: lib/WebService/Braintree/Xml.pm [hunk 2]
```diff
@@ -2,14 +2,15 @@ package WebService::Braintree::Xml;
...
-use vars qw(@ISA @EXPORT @EXPORT_OK %EXPORT_TAGS );
+use vars qw(@ISA @EXPORT_OK);
 use Exporter;
 our @ISA = qw(Exporter);
-our @EXPORT = qw(hash_to_xml xml_to_hash array collect_from_array);
-our @EXPORT_OK = qw();
-use WebService::Braintree::Util;
+our @EXPORT_OK = qw(hash_to_xml xml_to_hash);
+
+use XML::Simple;
```
Comment (singingfish) [inline lib/WebService/Braintree/Xml.pm]: looks like XML::Simple is unused (thankfully)

**Comment by singingfish** [comment]
I reckon for the most part this looks like a decent first pass cleanup.  

There are the variable name problems, and the sandbox fails need some further investigation via bisect.

**Comment by robkinyon** [comment]
I'm closing this PR and re-opening it again so that the history is clean.

**Pull request closed without merging**